### PR TITLE
Clarify csrr[s|c]i write behavior when imm=0

### PR DESCRIPTION
**Pull request by andresag01**
Fixes https://github.com/riscv/riscv-cheri/issues/118

### CODE REVIEW

**Comment by jrtc27** [comment]
Changes make sense but please update the commit message as it sounds like you're saying that CSR[S|C]I are being clarified as having a register source operand!

**Comment by tariqkurd-repo** [comment]
I filed an issue for this to assist with tracking: https://github.com/riscv/riscv-cheri/issues/118

**Review by tariqkurd-repo — APPROVED** [review]

**Comment by andresag01** [comment]
> Changes make sense but please update the commit message as it sounds like you're saying that CSRR[S|C]I are being clarified as having a register source operand!

Sorry about that! I changed the commit title and hopefully the new one makes more sense.

File: src/insns/csrr_32bit.adoc [hunk 1]
```diff
@@ -63,18 +63,18 @@ xref:extended_CSR_writing[xrefstyle=short] for the action taken on writing an XL
 +
 If `cd` is `c0` (or `rd` is `x0`), then <<CSRRWI>> shall not read the CSR and
 and shall not cause any of the side effects that might occur on a CSR read. If
-`rs1` is `x0`, then <<CSRRS>>, <<CSRRC>>, <<CSRRSI>> and <<CSRRCI>> will not
-write to the CSR at all, and so shall not cause any of the side effects that
-might otherwise occur on a CSR write.
+`rs1` is `x0` for <<CSRRS>> and <<CSRRC>>, or `imm` is 0 for <<CSRRSI>> and
```
Comment (arichardson) [inline src/insns/csrr_32bit.adoc:66]:
```suggestion
`rs1` is `x0` for <<CSRRS>> and <<CSRRC>>, or `uimm` is 0 for <<CSRRSI>> and
```

Just looked at the unpriv spec and it uses `uimm`. Does not need to be fixed here, but should probably clean this up later.

**Review by arichardson — APPROVED** [review]

**Pull request merged**